tests/MongoDB.Driver.Examples/ChangeStreamExamples.cs

```diff
@@ -122,5 +126,90 @@ public void ChangeStreamExample3()
                 next.FullDocument.Should().Be(documents[1]);
             }
         }
+
+        [Fact]
+        public void Changestream_Example_4()
+        {
+            RequireServer.Check().Supports(Feature.AggregateAddFields);
+
+            var client = DriverTestConfiguration.Client;
+            var database = client.GetDatabase("ChangeStreamExamples");
+            database.DropCollection("inventory");
+            var inventoryCollection = database.GetCollection<BsonDocument>("inventory");
+
+            //const pipeline = [
+            //{ $match: { $or : [{ 'fullDocument.username': 'alice' }, { 'operationType' : 'delete' }] } },
+
+            //{ $addFields: { newField: 'this is an added field!' } }
+
+            //];
+            //const collection = db.collection('inventory');
+            //const changeStream = collection.watch(pipeline);
+            //const next = await changeStream.next();
+
+            var cancelationTokenSource = new CancellationTokenSource();
+            try
+            {
+                BsonDocument document = new BsonDocument("username", "alice");
+
+                Task.Run(() =>
+                {
+                    while (!cancelationTokenSource.IsCancellationRequested)
+                    {
+                        Thread.Sleep(TimeSpan.FromMilliseconds(500));
+                        inventoryCollection.InsertOne(document);
+                        inventoryCollection.DeleteOne(document);
+                    }
+                }, cancelationTokenSource.Token);
+
+                // Start Changestream Example 4
+                var pipeline = new EmptyPipelineDefinition<ChangeStreamDocument<BsonDocument>>()
```

Comment:
Weird to me that you'd create an EmptyPipelineDefinition and then make it non-empty.  @rstam is this correct usage?

Reply:
We discussed this with @rstam. 
For me it's ok, I consider an empty pipeline object as a skeleton which will grow by adding stages with some functionality.